lightning/src/chain/onchaintx.rs

```diff
@@ -772,8 +772,11 @@ impl<ChannelSigner: EcdsaChannelSigner> OnchainTxHandler<ChannelSigner> {
 			for j in 0..i {
 				if requests[i].can_merge_with(&requests[j], cur_height) {
 					let merge = requests.remove(i);
-					requests[j].merge_package(merge);
-					break;
+					if let Err(rejected) = requests[j].merge_package(merge) {
+						requests.insert(i, rejected);
```

Comment:
Yeah, the re-inserts were only introduced as an alternative to panicking on the `Err(_)` here. However, those errors should never occur as we call `can_merge_with` beforehand. Added a `debug_assert!(false, _)`. I can change the re-inserts to a `panic!` as well to maintain the previous behavior.

The main goal was to push `panic!`s up in the stack, while avoiding preconditions on `merge_package`. Since the `Result` of `merge_package` is determined by `can_merge_with`, the latter can be used beforehand to optimize any calls.

I can remove the commit as well though.

Reply:
Moved the fixup commit into the right place.